src/tests/resistance/handshake.rs

```diff
@@ -466,6 +464,140 @@ async fn r001_t8_HANDSHAKE_upgrade_req_field() {
+    let cfg = gen_cfg(format!("{:#9000}", "Ziggurat/1.0"));
+    assert!(!run_handshake_req_test_with_cfg(cfg, debug).await);
+}
+
+#[allow(non_snake_case)]
+#[tokio::test]
+async fn r001_t10_HANDSHAKE_closed_ledger_field() {
+    // ZG-RESISTANCE-001
+    // Expected valid value for the "Closed-ledger" field in the handshake should be a valid string.
+
+    let debug = Debug::disable();
+
+    let gen_cfg = |ledger: String| SynthNodeCfg {
+        handshake: Some(HandshakeCfg {
+            http_closed_ledger: Some(ledger),
+            ..Default::default()
+        }),
+        ..Default::default()
+    };
+
+    // Valid value for Closed-Ledger
+    let cfg = gen_cfg("X72fvYvkYwPj7iFsE4OTiSwSd5Okz40P+eBRwsOXo4g=".to_owned());
+    assert!(run_handshake_req_test_with_cfg(cfg, debug).await);
+
+    // Invalid values
+    // Too short
+    let cfg = gen_cfg("vYvkYwPj7iFsE4OTiSwSd5Okz40P+eBRwsOXo4g=".to_owned());
+    assert!(!run_handshake_req_test_with_cfg(cfg, debug).await);
+
+    // Too long
+    let cfg = gen_cfg("X72fvYvkYwPj7iFsE4OTiSwSdSwSd5Okz40P+eBRwsOXo4g=".to_owned());
+    assert!(!run_handshake_req_test_with_cfg(cfg, debug).await);
+
+    // Empty
+    let cfg = gen_cfg("".to_owned());
+    assert!(!run_handshake_req_test_with_cfg(cfg, debug).await);
+
+    // Valid encoding but too long
+    let cfg = gen_cfg(format!(
+        "{:#9000}",
+        "X72fvYvkYwPj7iFsE4OTiSwSd5Okz40P+eBRwsOXo4g="
+    ));
+    assert!(!run_handshake_req_test_with_cfg(cfg, debug).await);
+}
+
+#[allow(non_snake_case)]
+#[tokio::test]
+async fn r001_t11_HANDSHAKE_previous_ledger_field() {
+    // ZG-RESISTANCE-001
+    // Expected valid value for the "Previous-Ledger" field in the handshake should be a valid string.
+
+    let debug = Debug::disable();
+
+    let gen_cfg = |ledger: String| SynthNodeCfg {
+        handshake: Some(HandshakeCfg {
+            http_prev_ledger: Some(ledger),
+            ..Default::default()
+        }),
+        ..Default::default()
+    };
+
+    // Valid value for Previous-Ledger but unable to do handshake
+    let cfg = gen_cfg("6bsyOYDSAux+Ubqyqo41NT+ce9q1m/FzeOrdTQSG758=".to_owned());
+    assert!(!run_handshake_req_test_with_cfg(cfg, debug).await);
+
+    // Invalid values
+    // Empty
+    let cfg = gen_cfg("".to_owned());
+    assert!(!run_handshake_req_test_with_cfg(cfg, debug).await);
+}
+
+#[allow(non_snake_case)]
+#[tokio::test]
+async fn r001_t12_HANDSHAKE_extra_field() {
+    // ZG-RESISTANCE-001
+    // Checks for handshake status after sending various extra header fields.
+
+    let debug = Debug::disable();
+
+    let gen_cfg = |value: String| SynthNodeCfg {
+        handshake: Some(HandshakeCfg {
+            http_unexpected_extra_field_and_value: Some(value),
+            ..Default::default()
+        }),
+        ..Default::default()
+    };
+
+    // "Valid" fields
+    let cfg = gen_cfg("Exec: /bin/bash".to_owned());
+    assert!(run_handshake_req_test_with_cfg(cfg, debug).await);
```

Comment:
let's try to repeat one Header field here, like the Upgrade field or ident field

so the same header appears twice or something.

Reply:
I've changed this test to give "Connect-As" header with different value (comparing to the one that's always present) and surprisingly it passes. I guess we can report it in some way to the client.

Reply:
Nice,
but can we also keep the previous subtests:
```rust
    // "Valid" fields
    let cfg = gen_cfg("Exec: /bin/bash".to_owned());
    assert!(run_handshake_req_test_with_cfg(cfg, debug).await);

    let cfg = gen_cfg("Exec: ".to_owned());
    assert!(run_handshake_req_test_with_cfg(cfg, debug).await);

    let cfg = gen_cfg(format!("{:#7000}", "Exec: /bin/bash"));
    assert!(run_handshake_req_test_with_cfg(cfg, debug).await);

    // Invalid value (too long)
    let cfg = gen_cfg(format!("{:#9000}", "Exec: /bin/bash"));
```